example/appengine/app.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The demo app shows how to use the github package on Google App Engine.
+package demo
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/log"
+)
+
+func init() {
+	http.HandleFunc("/", handler)
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/" {
+		http.NotFound(w, r)
+		return
+	}
+
+	ctx := appengine.NewContext(r)
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: os.Getenv("GITHUB_AUTH_TOKEN")},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
```

Comment:
This was expedient for this demo, but I believe it is possible to make a single client as a package global during initialization.

One caveat is that you can't call `os.Getenv` on App Engine within an `init` function due to the environment not being fully passed-in on app startup. The details are quite hairy.

So in cases like that, you could have a `sync.Once` block get the environment variable value and create the client before its first use... but I wanted to have a simple example here and not complicate things too much.